pyproject.toml

```diff
@@ -0,0 +1,76 @@
+[project]
+classifiers=[
+  "Operating System :: OS Independent",
+  "Intended Audience :: Science/Research",
+  "Programming Language :: Python",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  'Operating System :: Microsoft :: Windows',
+  'Operating System :: Unix',
+  'Operating System :: MacOS',
+  "Topic :: Scientific/Engineering",
+]
+
+dynamic = ["version"]
+keywords = ["unstructured grids"]
+description = "Xarray-styled package for reading and directly operating on unstructured grid datasets following UGRID conventions"
+license = {text = "Apache-2.0"}
+name = "uxarray"
+readme = "README.md"
+requires-python = ">=3.9"
+
+# minimal dependencies start
+dependencies = [
+  "xarray",
+  "cartopy",
+  "dask",
+  "numba",
+  "numpy",
+  "scikit-learn",
+  "scipy",
+  "pyarrow",
+  "pandas<2.1.0",
+  "netcdf4",
+  "requests"
+]
+# minimal dependencies end
+
+[project.optional-dependencies]
+complete = ["uxarray[viz, math, dev]"]
+viz = [
+  'datashader',
+  'holoviews',
+  'cartopy',
+  'geoviews',
+  'antimeridian',
+  'matplotlib-base',
+  'matplotlib-inline',
+  'shapely',
+  'spatialpandas'
```

Comment:
So, how will these optional dependencies be handled by PyPI in the release? I see that this is similar to how Xarray handles theirs but curious about the execution of this.

Reply:
https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies

I've updated our installation guide to reflect this. 

https://uxarray--771.org.readthedocs.build/en/771/installation.html#installing-uxarray-via-pypi